packages/plugin-aws-lambda/src/index.js

```diff
@@ -26,6 +26,14 @@ function wrapHandler (client, flushTimeoutMs, handler) {
   }
 
   return async function (event, context) {
+    const startTime = new Date()
+
+    client.addOnError(event => {
+      const endTime = new Date()
+
+      event.app.duration = endTime - startTime
```

Comment:
Presumably setting this here means that it overwrites the value set by the `plugin-app-duration` plugin? In that case all other `onError` callbacks supplied by the user will see the yet-to-be-overwritten value.

That might be a tolerable side-effect of keeping this implementation simple.

The "in pursuit of full correctness" solution, for comparison might look something like this:

`plugin-app-duration/app.js`
```js
let appStart
const reset = () => { appStart = new Date() }
reset()

module.exports = {
  name: 'appDuration',
  load: client => {
    client.addOnError(event => {
      const now = new Date()

      event.app.duration = now - appStart
    }, true)

    return { reset }
  }
}
```

`plugin-aws-lambda/src/index.js`
```js
// ...
  return async function (event, context) {
    client.getPlugin('appDuration').reset()
    // ...
  })
```

Reply:
Yeah, I think your way makes more sense. It's probably also more accurate to call `reset` in the `load` method itself rather than the lambda wrapper